packages/devextreme/testing/testcafe/tests/dataGrid/filterPanel.ts

```diff
@@ -0,0 +1,41 @@
+import { Selector } from 'testcafe';
+import url from '../../helpers/getPageUrl';
+import createWidget from '../../helpers/createWidget';
+import DataGrid from '../../model/dataGrid';
+
+fixture.disablePageReloads`filterPanel`
+  .page(url(__dirname, '../container.html'));
+
+const POPUP_CLASS = '.dx-popup';
+const INVISIBLE_CLASS = '.dx-state-invisible';
+
+function getVisiblePopups(): Selector {
+  return Selector(`${POPUP_CLASS}:not(${INVISIBLE_CLASS})`);
+}
+
+// T1182854
+test('editor\'s popup inside filterBuilder is opening & closing right (T1182854)', async (t) => {
+  const dataGrid = new DataGrid('#container');
+
+  const filterBuilder = (
+    await dataGrid.getFilterPanel().openFilterBuilderPopup(t)
+  ).getFilterBuilder();
+
+  await t.expect(getVisiblePopups().count).eql(1); // only filterPanel popup
+
+  await t.click(filterBuilder.getField().getValueText());
+  await t.expect(getVisiblePopups().count).eql(2); // filterPanel popup + editor popup
+
+  await t.click(filterBuilder.getField().getValueText());
+  await t.expect(getVisiblePopups().count).eql(1); // only filterPanel popup
+
+  await t.click(filterBuilder.getField().getValueText());
+  await t.expect(getVisiblePopups().count).eql(2); // filterPanel popup + editor popup
```

Comment:
Maybe use a visual test instead? Multiple popups can be visible at the same time, and a popup can overlap other popups. For instance, the editor popup can be hidden behind the filterPanel popup although both popups are visible. Though, I understand that it will not happen in this usage scenario.

Reply:
I'm afraid visual test could be unstable, so I think for now it's better to leave this test as is

Reply:
Why do think test can be unstable?